src/core/connected_clients.rs

```diff
@@ -0,0 +1,45 @@
+use bevy::prelude::*;
+
+use crate::core::ClientId;
+
+#[derive(Resource, Default)]
+pub struct ConnectedClients {
+    clients: Vec<ClientId>,
+    replicate_after_connect: bool,
+}
+
+impl ConnectedClients {
+    pub(crate) fn new(replicate_after_connect: bool) -> Self {
+        Self {
+            clients: Default::default(),
+            replicate_after_connect,
+        }
+    }
+
+    /// Gets if clients will automatically have replication enabled for them after they connect.
+    pub fn replicate_after_connect(&self) -> bool {
+        self.replicate_after_connect
+    }
+
+    /// Gets an iterator over all [`ClientId`]s currently tracked.
```

Comment:
It's preferred to use 'returns' for consistency with the Rust standard library.
```suggestion
    /// Returns if clients will automatically have replication enabled for them after they connect.
    pub fn replicate_after_connect(&self) -> bool {
        self.replicate_after_connect
    }

    /// Returns an iterator over all [`ClientId`]s currently tracked.
```